### PR TITLE
#50 dev

### PR DESCRIPTION
**Pull request by gaticosalas**
Route for username search

### CODE REVIEW

File: routes/api/users.js [hunk 1]
```diff
@@ -83,4 +83,33 @@ router.post('/', [
...
+// @desc   Gets user who's username contains the input
+// @access Public
+router.get('/search', [], async (req, res ) => {
+    const { input } = req.body;
+    const regex = new RegExp(`${input}`, 'i');
+
+    try{
+        let users_array = await User.find({
+            username: regex
+        }, '_id username avatar').limit(10).exec();
```
Comment (loren589) [inline routes/api/users.js:97]: Just out of curiosity, is the 10 limit there for search results? As in, 10 results are shown when you search by username?

**Review by loren589 — APPROVED** [review]
Looks really good!

**Review by MKB-1 — APPROVED** [review]
Good job optimizing the search to reduce strain on the server

File: routes/api/users.js (hunk 1 above)
Comment (gaticosalas) [reply routes/api/users.js:97]: @loren589 exactly! Mongoose has a lot of different ways you can query your search, if ever you're interested, you can look up the mongoose docs.

**Pull request merged**